include/envoy/network/socket.h

```diff
@@ -242,10 +242,11 @@ class SocketInterface {
    * @param type type of socket requested
    * @param addr_type type of address used with the socket
    * @param version IP version if address type is IP
+   * @param socket_v6only if the socket is ipv6 version only
    * @return @ref Network::IoHandlePtr that wraps the underlying socket file descriptor
    */
-  virtual IoHandlePtr socket(Socket::Type type, Address::Type addr_type,
-                             Address::IpVersion version) PURE;
+  virtual IoHandlePtr socket(Socket::Type type, Address::Type addr_type, Address::IpVersion version,
+                             int socket_v6only) PURE;
```

Comment:
I think the interface should use a bool instead of int here (even though it needs to be converted to int for `setsockopt`).

Reply:
fair point it has been fixed in 0c2558e59aa266db6042d81707f9e439684424de